solidity/contracts/converter/types/liquidity-pool-v3/LiquidityPoolV3Converter.sol

```diff
@@ -0,0 +1,1356 @@
+        reserveTokens.push(_token);
+        reserveIds[_token] = reserveTokens.length;
+    }
+
+    /**
+      * @dev returns the reserve's weight
+      * added in version 28
+      *
+      * @param _reserveToken    reserve token contract address
+      *
+      * @return reserve weight
+    */
+    function reserveWeight(IERC20Token _reserveToken)
+        public
+        view
+        validReserve(_reserveToken)
+        returns (uint32)
+    {
+        return PPM_RESOLUTION / 2;
+    }
+
+    /**
+      * @dev returns the reserve's balance
+      * note that prior to version 17, you should use 'getConnectorBalance' instead
+      *
+      * @param _reserveToken    reserve token contract address
+      *
+      * @return reserve balance
+    */
+    function reserveBalance(IERC20Token _reserveToken)
+        public
+        override
+        view
+        returns (uint256)
+    {
+        uint256 reserveId = reserveIds[_reserveToken];
+        require(reserveId != 0, "ERR_INVALID_RESERVE");
+        return getReserveBalance(reserveId);        
+    }
+
+    /**
+      * @dev converts a specific amount of source tokens to target tokens
+      * can only be called by the bancor network contract
+      *
+      * @param _sourceToken source ERC20 token
+      * @param _targetToken target ERC20 token
+      * @param _amount      amount of tokens to convert (in units of the source token)
+      * @param _trader      address of the caller who executed the conversion
+      * @param _beneficiary wallet to receive the conversion result
+      *
+      * @return amount of tokens received (in units of the target token)
+    */
+    function convert(IERC20Token _sourceToken, IERC20Token _targetToken, uint256 _amount, address _trader, address payable _beneficiary)
+        public
+        override
+        payable
+        protected
+        only(BANCOR_NETWORK)
+        returns (uint256)
+    {
+        // validate input
+        require(_sourceToken != _targetToken, "ERR_SAME_SOURCE_TARGET");
+
+        // if a whitelist is set, verify that both and trader and the beneficiary are whitelisted
+        require(address(conversionWhitelist) == address(0) ||
+                (conversionWhitelist.isWhitelisted(_trader) && conversionWhitelist.isWhitelisted(_beneficiary)),
+                "ERR_NOT_WHITELISTED");
+
+        return doConvert(_sourceToken, _targetToken, _amount, _trader, _beneficiary);
+    }
+
+    /**
+      * @dev returns the conversion fee for a given target amount
+      *
+      * @param _targetAmount  target amount
+      *
+      * @return conversion fee
+    */
+    function calculateFee(uint256 _targetAmount) internal view returns (uint256) {
+        return _targetAmount.mul(conversionFee) / PPM_RESOLUTION;
+    }
+
+    /**
+      * @dev gets the stored reserve balance for a given reserve id
+      *
+      * @param _reserveId   reserve id
+    */
+    function getReserveBalance(uint256 _reserveId) internal view returns (uint256) {
```

Comment:
we don't use `get` for getters in our convention - can just replace with overloading.

Reply:
That's not a public getter, and overloading `reserveBalance` would be confusing IMO.

Reply:
we don't use it for non public functions either

Reply:
We actually have 63 of those (most of which are **not** deprecated), scattered across 10 different files.
So while I don't mind doing this, I don't quite see the point.
The convention that we've established was for functions which return private or internal variables, in order to be consistent with auto-generated functions for public variables.
This function does not return some private or internal variable.

Reply:
Fixed.